src/envoy/mixer/http_control.cc

```diff
@@ -91,47 +118,50 @@ std::string GetLastForwardedHost(const HeaderMap& header_map) {
   return xff_list.back();
 }
 
-void FillRequestHeaderAttributes(const HeaderMap& header_map,
-                                 Attributes* attr) {
-  // Pass in all headers
+std::map<std::string, std::string> ExtractHeaders(const HeaderMap& header_map) {
+  std::map<std::string, std::string> headers;
   header_map.iterate(
       [](const HeaderEntry& header, void* context) {
-        Attributes* attr = static_cast<Attributes*>(context);
-        attr->attributes[kRequestHeaderPrefix + header.key().c_str()] =
-            StringValue(header.value().c_str());
+        std::map<std::string, std::string>* header_map =
+            static_cast<std::map<std::string, std::string>*>(context);
+        (*header_map)[header.key().c_str()] = header.value().c_str();
       },
-      attr);
+      &headers);
+  return headers;
+}
 
-  SetStringAttribute(kAttrNameRequestPath, header_map.Path()->value().c_str(),
-                     attr);
-  SetStringAttribute(kAttrNameOriginIp, GetFirstForwardedFor(header_map), attr);
-  SetStringAttribute(kAttrNameOriginHost, GetLastForwardedHost(header_map),
-                     attr);
+void FillRequestHeaderAttributes(const HeaderMap& header_map,
+                                 Attributes* attr) {
+  SetStringAttribute(kRequestPath, header_map.Path()->value().c_str(), attr);
+  SetStringAttribute(kRequestHost, header_map.Host()->value().c_str(), attr);
+  attr->attributes[kRequestTime] = TimeValue(std::chrono::system_clock::now());
+  SetStringAttribute(kOriginIp, GetFirstForwardedFor(header_map), attr);
```

Comment:
IP addresses should be encoded as byte arrays over the wire, 4 bytes for ipv4 and 16 bytes for ipv6